File: map-sum-pairs/map_sum_pairs.js
/**
 * Initialize your data structure here.
 */
var MapSum = function() {
    this.d = new Array();
};

/** 
 * @param {string} key 
 * @param {number} val
 * @return {void}
 */
MapSum.prototype.insert = function(key, val) {
    this.d[key] = val;
};

/** 
 * @param {string} prefix
 * @return {number}
 */
MapSum.prototype.sum = function(prefix) {
    var sum = 0;
    var sLen = prefix.length;
    for (var key in this.d) {
        if (key.slice(0, sLen) == prefix) {
            sum += this.d[key];
        }
    }
    return sum
};

/** 
 * Your MapSum object will be instantiated and called as such:
 * var obj = Object.create(MapSum).createNew()
 * obj.insert(key,val)
 * var param_2 = obj.sum(prefix)
 */

File: map-sum-pairs/map_sum_pairs.go
type MapSum struct {
    m   map[string]int
}


/** Initialize your data structure here. */
func Constructor() MapSum {
    return MapSum{m: make(map[string]int)}
}


func (this *MapSum) Insert(key string, val int)  {
    this.m[key] = val
}


func (this *MapSum) Sum(prefix string) int {
    var sum int
    s_len := len(prefix)
    for k, v := range this.m {
        if k[0:s_len] == prefix {
            sum += v
        }
    }
    return sum
}


/**
 * Your MapSum object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Insert(key,val);
 * param_2 := obj.Sum(prefix);
 */